### DISCUSSION TITLE
MIDI_CREATE_DEFAULT_INSTANCE fails for Arduino Nano

### DISCUSSION THREAD

**Discussion by glosterva** [FortySevenEffects/arduino_midi_library#202] [Q&A] [answered]
My program starts with this:

```c++
#include <Midi.h>
// Create and bind the MIDI interface to the default hardware Serial port
 MIDI_CREATE_DEFAULT_INSTANCE();
```

It gets this error, with a carat under the semicolon at the end:
```
MidiRibbonV1:4:31: error: expected constructor, destructor, or type conversion before ';' token MIDI_CREATE_DEFAULT_INSTANCE();
                               ^
```
So I comment that out, and insert what the macro would be putting in:
```c++
MIDI_CREATE_INSTANCE(HardwareSerial, Serial,  MIDI);
```
This results in these errors.  In the first error, the carat points at the second parameter, Serial. In the second error. it points at the third parameter, MIDI. In the third error, it points to the semicolon at the end.
```
MidiRibbonV1:8:38: error: 'Serial' is not a type
 MIDI_CREATE_INSTANCE(HardwareSerial, Serial,  MIDI);
                                      ^~~~~~
MidiRibbonV1:8:47: error: 'MIDI' has not been declared
 MIDI_CREATE_INSTANCE(HardwareSerial, Serial,  MIDI);
                                               ^~~~
MidiRibbonV1:8:52: error: expected constructor, destructor, or type conversion before ';' token
 MIDI_CREATE_INSTANCE(HardwareSerial, Serial,  MIDI);
                                                    ^
```
On the Nano, Serial is the name of the serial port (as you can see on [this page](https://www.arduino.cc/reference/en/language/functions/communication/serial/)) as it has only one serial port.
I have installed version 5.0.2 of the MIDI library and am using the latest Arduino IDE, version 1.8.13.  Do you have suggestions on how can I get this to work?  Thanks!
-Vance

**Comment by franky47** [comment, marked as answer]
Can you try with `#include <MIDI.h>` (uppercase)?

It could be that the macro is not found (on case-sensitive systems) and it throws the compiler off.

**Reply by glosterva** [reply]
Thanks!  That worked well.
